src/cil.ml

```diff
@@ -2839,9 +2859,9 @@ let parseInt (str: string) : exp =
   (* The length of the suffix and a list of possible kinds. See ISO
   * 6.4.4.1 *)
   let hasSuffix = hasSuffix str in
-  let suffixlen, kinds =
+  let suffixlen, kinds = (* TODO check situation with __int128 *)
     if hasSuffix "ULL" || hasSuffix "LLU" then
-      3, [IULongLong]
+      3, [IULongLong] (* ? [if !M.theMachine.M.sizeof_longlong = 16 then IUInt128 else IULongLong] *)
```

Comment:
Yes, that's what I hinted at above.
Both would fit, but `__int128` is more concrete since it does no longer depend on the machine.
Idk if that might be of value for something - probably not unless we do it for all machine-dependent types.
So, yes, keeping the original type makes more sense at this point.